Kommunicate/Classes/KMConversationListViewController.swift

```diff
@@ -0,0 +1,571 @@
+                guard let alert = alertValue as? String else { return }
+                let alertComponents = alert.components(separatedBy: ":")
+                if alertComponents.count > 1 {
+                    message.message = alertComponents[1]
+                } else {
+                    message.message = alertComponents.first
+                }
+                weakSelf.viewModel.addMessages(messages: [message])
+            } else if updateUI == Int(APP_STATE_INACTIVE.rawValue) {
+                // Coming from background
+
+                guard contactId != nil || groupId != nil || conversationId != nil else { return }
+                weakSelf.launchChat(contactId: contactId, groupId: groupId, conversationId: conversationId)
+            }
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "reloadTable"), object: nil, queue: nil, using: { [weak self] notification in
+            NSLog("Reloadtable notification received")
+
+            guard let weakSelf = self, let list = notification.object as? [Any] else { return }
+            weakSelf.viewModel.updateMessageList(messages: list)
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "USER_DETAILS_UPDATE_CALL"), object: nil, queue: nil, using: { [weak self] notification in
+            NSLog("update user detail notification received")
+
+            guard let weakSelf = self, let userId = notification.object as? String else { return }
+            print("update user detail")
+            weakSelf.viewModel.updateUserDetail(userId: userId) { (success) in
+                if success {
+                    weakSelf.tableView.reloadData()
+                }
+            }
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "UPDATE_CHANNEL_NAME"), object: nil, queue: nil, using: { [weak self] _ in
+            NSLog("update group name notification received")
+            guard let weakSelf = self, weakSelf.view.window != nil else { return }
+            print("update group detail")
+            weakSelf.tableView.reloadData()
+        })
+    }
+
+
+     func removeObserver() {
+        if alMqttConversationService != nil {
+            alMqttConversationService.unsubscribeToConversation()
+        }
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "pushNotification"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "newMessageNotification"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "reloadTable"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "USER_DETAILS_UPDATE_CALL"), object: nil)
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "UPDATE_CHANNEL_NAME"), object: nil)
```

Comment:
We can define these constants from where these notifications are posted(AL framework) and use that.